### PR TITLE
Introduce getAllWorkflowPods and getRunningWorkflowPods.

### PR DESCRIPTION
**Pull request by shrinandj**

### CODE REVIEW

**Comment by sandeepbhojwani** [comment]
What's wrong with the existing code?

File: workflow/controller/operator.go [hunk 1]
```diff
@@ -283,7 +283,7 @@ func (woc *wfOperationCtx) podReconciliation() error {
 	// including Pending pods in the query. If one of our nodes does not show up in this list,
```
Comment (shrinandj) [inline workflow/controller/operator.go]: I just realized that this comments needs to be updated. Will do.

**Comment by shrinandj** [comment]
The current code is hard to understand. The call sites in `podReconciliation` need a list of running pods and a list of all pods associated with the workflow reflectively. Therefore, having two different functions makes it easy to understand.

**Comment by shrinandj** [comment]
As mentioned earlier, in `podReconciliation`, there are 2 places where the list of pods is needed. One place needs all running pods and another place needs all pods (running or not). This code addresses these two case by having to explicit functions that do that making it easier to reason about the code.

**Comment by jessesuen** [comment]
But the logic is now incorrect. In no circumstance should the controller be examining pods it already marked completed (which this change does in getAllWorkflowPods()). It's an invariant that that the controller should only ever operate on pods that have completed=false. 

Second, by omitting `status.phase!=Pending` filter from getRunningWorkflowPods(), this causes podReconciliation and K8s API server to do more work than necessary because it causes podReconciliation to examine pending pods, of which there is no action to be done. These optimizations were added during scalability testing and I am not comfortable removing those filters.

**Pull request closed without merging**